crates/bevy_editor_core/src/selection.rs

```diff
@@ -0,0 +1,86 @@
+//! Editor selection module.
+
+use bevy::{ecs::entity::Entities, prelude::*};
+
+/// Editor selection plugin.
+#[derive(Default)]
+pub struct SelectionPlugin;
+
+impl Plugin for SelectionPlugin {
+    fn build(&self, app: &mut App) {
+        app.init_resource::<SelectedEntity>()
+            .register_type::<SelectedEntity>()
+            .add_systems(PostUpdate, reset_selected_entity_if_entity_despawned);
+    }
+}
+
+/// The currently selected entity in the scene.
+#[derive(Resource, Default, Reflect)]
+#[reflect(Resource, Default)]
+pub struct SelectedEntity(pub Option<Entity>);
+
+impl SelectedEntity {
+    /// Toggle selection for an entity.
+    pub fn toggle(&mut self, entity: Entity) {
+        debug_assert_ne!(entity, Entity::PLACEHOLDER);
+        if self.0 == Some(entity) {
+            self.0 = None;
+        } else {
+            self.0 = Some(entity);
+        }
+    }
+
+    /// Set an entity as selected.
+    pub fn set(&mut self, entity: Entity) {
+        debug_assert_ne!(entity, Entity::PLACEHOLDER);
+        self.0 = Some(entity);
+    }
+
+    /// Empty the selection.
+    pub fn reset(&mut self) {
+        self.0 = None;
+    }
+}
+
+/// System to removes entity [`SelectedEntity`] when the entity is despawned.
```

Comment:
```suggestion
/// System to reset [`SelectedEntity`] when the entity is despawned.
```

Reply:
Oops, must have tried rewriting that at some point. Thanks!